clang/test/BoundsSafety/Frontend/bounds-safety-bringup-missing-chekcs-disabled.c

```diff
@@ -0,0 +1,103 @@
+
+// RUN: %clang_cc1 -fbounds-safety -fbounds-safety-bringup-missing-checks=all   -fno-bounds-safety-bringup-missing-checks=libc_attributes -fsyntax-only %s -verify=libca-off
+// libca-off-warning@*{{compiling with "libc_attributes" bounds check disabled is deprecated; compile with -fbounds-safety-bringup-missing-checks=batch_0 to enable the new bound checks}}
+
+// RUN: %clang_cc1 -fbounds-safety -fbounds-safety-bringup-missing-checks=all   -fno-bounds-safety-bringup-missing-checks=array_subscript_agg -fsyntax-only %s -verify=asa-off
+// asa-off-warning@*{{compiling with "array_subscript_agg" bounds check disabled is deprecated; compile with -fbounds-safety-bringup-missing-checks=batch_0 to enable the new bound checks}}
+
+// =============================================================================
+// two checks off
+// =============================================================================
+
+// Don't be exhaustive to keep this test case smaller
+
+// RUN: %clang_cc1 -fbounds-safety -fbounds-safety-bringup-missing-checks=all \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=access_size \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=indirect_count_update \
+// RUN:   -fsyntax-only %s -verify=as-off,icu-off
+
+// RUN: %clang_cc1 -fbounds-safety -fbounds-safety-bringup-missing-checks=all \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=access_size \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=return_size \
+// RUN:   -fsyntax-only %s -verify=as-off,rs-off
+
+// RUN: %clang_cc1 -fbounds-safety -fbounds-safety-bringup-missing-checks=all \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=access_size \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=ended_by_lower_bound \
+// RUN:   -fsyntax-only %s -verify=as-off,eblb-off
+
+// RUN: %clang_cc1 -fbounds-safety -fbounds-safety-bringup-missing-checks=all \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=access_size \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=compound_literal_init \
+// RUN:   -fsyntax-only %s -verify=as-off,cli-off
+
+// RUN: %clang_cc1 -fbounds-safety -fbounds-safety-bringup-missing-checks=all \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=access_size \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=libc_attributes \
+// RUN:   -fsyntax-only %s -verify=as-off,libca-off
+
+// RUN: %clang_cc1 -fbounds-safety -fbounds-safety-bringup-missing-checks=all \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=access_size \
+// RUN:   -fno-bounds-safety-bringup-missing-checks=array_subscript_agg \
+// RUN:   -fsyntax-only %s -verify=as-off,asa-off
+
+
+// =============================================================================
+// Check the diagnostics can be made into errors
+// =============================================================================
+
+// FIXME: This doesn't work right now because
+// `TextDiagnosticBuffer::FlushDiagnostics` doesn't preserve diagnostic IDs when
```

Comment:
@rapidsna @hnrklssn @patrykstefanski It turns diagnostics don't work properly where we currently emit diagnostics. In particular

* `-Werror` doesn't work. The warning is still treated as a warning
* The diagnostic printer doesn't show the warning group
* `-Wno-bounds-safety-legacy-checks-enabled` doesn't suppress the warning.

We emit a bunch of `-fbounds-safety` diagnostic in this location which will all have the same problem.

I've figured out the problem but fixing it is very much of of scope for this patch.

Reply:
I ended moving the emission of the diagnostics out of CC1 and into the driver so this isn't a problem anymore.